captain/parser/bool_parser/bool_parser.py

```diff
@@ -0,0 +1,250 @@
+import string
+from typing import Union, List, Type
+from bool_parser.types.expressions import Token
+from functools import partial
+
+from bool_parser.expressions.models import (
+    PARANTHESES_TO_CLASS,
+    BooleanLiteral,
+    NumericLiteral,
+    LeftParanthesis,
+    Operator,
+    ReturnTypes,
+    RightParanthesis,
+    SymbolTableType,
+)
+from bool_parser.expressions.exceptions import (
+    InvalidCharacter,
+    InvalidExpression,
+    InvalidIdentifier,
+    MatchError,
+    MissingLeftParanthesis,
+    MissingRightParanthesis,
+)
+from bool_parser.expressions.models import (
+    Expression,
+    Identifier,
+    TOKEN_TO_CLASS,
+)
+from bool_parser.parser_config import rules
+
+parantheses = set(PARANTHESES_TO_CLASS.keys())
+operation_symbols = set(TOKEN_TO_CLASS.keys())
+variable_symbols = set(string.ascii_letters + string.digits + "_./")
+boolean_literal_symbols = set("TrueFalse")
+numeric_literal_symbols = set(string.digits + ".")
+language = {
+    *parantheses,
+    *boolean_literal_symbols,
+    *operation_symbols,
+    *variable_symbols,
+}
+
+ParseItem = Union[Token, Expression]
+
+
+def _tokenize(s: str, symbol_table: SymbolTableType) -> list[Token]:
+    """
+    Tokenizes the string input from the user
+    """
+
+    def _match_op_symbol(ptr: int) -> tuple[Operator, int]:
+        extend = 0
+        while (
+            ptr + extend + 1 <= len(s)
+            and s[ptr : ptr + extend + 1] in TOKEN_TO_CLASS.keys()
+        ):
+            extend += 1
+        c = s[ptr : ptr + extend]
+        return Operator(c), extend
+
+    def _match_literal_or_var(ptr: int, allowed_symbols: set[str]):
+        start = ptr
+        while ptr < len(s) and s[ptr] in allowed_symbols:
+            ptr += 1
+        if ptr < len(s) and s[ptr] not in language:
+            raise InvalidCharacter
+        return start, ptr
+
+    tokens: List[Token] = []
+    s = s.replace(" ", "")  # remove whitespace
```

Comment:
Potential problem with files that have space in the name ?

Reply:
I think we can safely assume test files will not have whitespace? not sure tho, could have smthg to check it

Reply:
Agree with @LatentDream, we should handle that properly, otherwise this will just be a bug for the future.

Reply:
replaced underscore with whitespace and added unique name checking